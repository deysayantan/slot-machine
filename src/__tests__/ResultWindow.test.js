import React from 'react';
import ReactDOM from 'react-dom';
import ResultWindowElement from '../components/slot/ResultWindow';
import Adapter from 'enzyme-adapter-react-16'
import { configure,shallow,mount } from 'enzyme'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";


const mockStore = configureMockStore();
const store = mockStore({});
const mockCallBack = jest.fn(); 
configure({ adapter: new Adapter() })

const resultWindowElement = shallow((<Provider store={store}><ResultWindowElement /></Provider>));

describe('ResultWindow  snapshot testing',()=>{
    it('renders app module correctly',()=>{
        expect(resultWindowElement).toMatchSnapshot();
    });
    it('Result should not appear immediately after first load',()=>{
        const componentInstance = resultWindowElement.instance();
        const resultWindow = new componentInstance.props.children.type.WrappedComponent(0)
        expect(resultWindow.state.shouldAppear).toEqual(false)
    })
});
