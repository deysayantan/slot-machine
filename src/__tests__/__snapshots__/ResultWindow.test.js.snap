// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultWindow  snapshot testing renders app module correctly 1`] = `ShallowWrapper {}`;
